src/utils/alignment-D3.js

```diff
@@ -84,56 +55,59 @@ export function areShapesAlignedWithTicks(
-    getTickValue(ticksCollection[0], dataType);
-
+  const increment = tickValues[1] - tickValues[0];
+  tickValues = [
+    tickValues[0] - increment,
+    ...tickValues,
+    tickValues[tickValues.length - 1] + increment
+  ];
+
+  let tickPositions = [].map.call(
+    ticksCollection,
+    tick => getTickPosition(tick)[coord]
+  );
+  const normalPositionOrder = tickPositions[1] > tickPositions[0];
   // positionIncrement may be positive or negative based on axis sort order
-  const positionIncrement = getTickPosition( ticksCollection[1] )[coord] -
-    getTickPosition( ticksCollection[0] )[coord];
+  const positionIncrement = tickPositions[1] - tickPositions[0];
+  tickPositions = [
+    tickPositions[0] - positionIncrement,
+    ...tickPositions,
+    tickPositions[tickPositions.length - 1] + positionIncrement
+  ];
 
   shapeCollection.forEach(function(shape) {
     let pos = getShapePosition(shape, dimension, positionType);
     let val = getShapeValue(shape, dataAttribute, dataType);
     // index is initially off (either -1 or ticksCollection.length), then
     // _getSurroundingTicks increments or decrements according to sort order
-    let tickIndex = ( !normalTickOrder ? ticksCollection.length : -1 );
 
     let surroundingTicks = _getSurroundingTicks(
-      shape,
       val,
-      ticksCollection,
-      tickIndex,
-      increment,
-      dataType,
-      normalTickOrder
+      tickValues,
+      normalValueOrder
     );
 
     if (surroundingTicks.length > 0) {
-      let tickPos0 = ( !surroundingTicks[0] ?
-        // surroundingTicks[0] may be null. if so, calc mock begin position
-        ( getTickPosition( surroundingTicks[1] )[coord] - positionIncrement ) :
-        // actual begin position
-        getTickPosition( surroundingTicks[0] )[coord]
-      );
+      let prevTick, nextTick;
+      if (normalPositionOrder) {
+        [prevTick, nextTick] = surroundingTicks;
+      } else {
+        [nextTick, prevTick] = surroundingTicks;
+      }
 
-      let tickPos1 = ( !surroundingTicks[1] ?
-        // calc mock end position
-        ( getTickPosition( surroundingTicks[0] )[coord] + positionIncrement ) :
-        // actual position
-        getTickPosition( surroundingTicks[1] )[coord]
-      );
-      let beforeTickPos = Math.min(tickPos0, tickPos1),
-        afterTickPos = Math.max(tickPos0, tickPos1);
+      let prevPos = tickPositions[prevTick];
+      let nextPos = tickPositions[nextTick];
 
-      // if shape is positioned between the two ticks plus or minus 11px
-      // TODO reduce to 2px after Bar Chart is fixed. A leeway is necessary
-      // for this code to work on all chart types.
-      if ( (pos >= beforeTickPos - 11 ) && ( pos <= afterTickPos + 11 ) ) {
+      // If shape is positioned between the two ticks plus or minus 2px
+      // A leeway is necessary for this code to work on all chart types.
+      if ( prevPos - 2 <= pos && pos <= nextPos + 2 ) {
```

Comment:
@tchaffee Can you please swap out the fCC Codepen code with that provided by @ValeraS in this PR? Either that or this needs to be changed to 11 to get the official pen to pass. Additionally, I ran the tests against a few other projects and found that 3px was needed, if that's ok with you to change, @ValeraS  .

Reply:
@tbushman Can I see these projects?

Reply:
https://codepen.io/deepikag/pen/KeGXLN

Reply:
@tbushman I have gotten next results:
- for the first pen, all tests pass in my environment. (max variation - 0.5)
- the second one is buggy and doesn't pass the tests with a current CDN bundle. After fixing xScale, the alignment test passes. (max variation - 0)

to calculate max variation:
```javascript
      if (pos < prevPos) {
        maxVariation = Math.max(maxVariation, prevPos - pos);
      } else if (pos > nextPos) {
        maxVariation = Math.max(maxVariation, pos - nextPos);
      }
```

Reply:
@ValeraS Thanks for checking. The first linked Codepen above is the only one not passing -- in Firefox. It passes with 3px. Thanks.

Reply:
I can't reproduce this issue, but I have increased a lag.